### ISSUE TITLE
N-API : napi_call_threadsafe_function isn't properly working when page is loaded by "loadURL"

### ISSUE THREAD

**Issue by orange4glace** [electron/electron#15144]
Electron version : 4.0.0-beta.2
Operating system : Windows 10 x64

**Expected Behavior**
napi_threadsafe_function should be called WHEN uv queue is free

**Actual behavior**
napi_thread_function is called only once or never even uv queue is free

**To Reproduce**
Here's google drive link for download
https://drive.google.com/open?id=1BRxUXs2OyqwjvX9oE-f7eilrFamDnfr7
(link updated)

To Reproduce:
1. install packages
`$npm install`
2. Build native module. For example,
`$HOME=~/.electron-gyp node-gyp configure build --target=4.0.0-beta.2 --arch=x64 --dist-url=https://atom.io/download/electron`
3. Start web server via `simple-server`
`$npm run serv`
4. Start electron
`$npm start`

Now you see your **TERMINAL** console prints some text for every 2 seconds, and at that time, native module calls napi_threadsafe_function but your **BROWSER** developer console prints nothing

**(Updated) Weird behavior**
When you refresh the page, it suddenly works properly (napi_threadsafe_function is called periodically).

To see the expected behavior,

Open `main.js` and replace 9th line with 

`win.loadFile('index.html')`

Now electron loads a page from local instead of web server.
Restart electron, now you see napi_threadsafe_function works correctly.

**Comment by jrd-rocks** [comment]
I've ran in this bug as well, is there a fix?

**Comment by electron-triage** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron 6.x.y or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons. This issue will be closed 7 days from now if there is no response.

Thanks in advance! Your help is appreciated.

**Comment by electron-triage** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
